aten/src/ATen/native/RNN.cpp

```diff
@@ -336,12 +337,13 @@ struct FullLayer : Layer<Tensor, hidden_type, cell_params> {
   FullLayer(Cell<hidden_type, cell_params>& cell)
     : cell_(cell) {};
 
-  unstacked_output_type operator()(std::vector<Tensor> step_inputs, const hidden_type& input_hidden, const cell_params& params) const {
-    std::vector<Tensor> step_outputs;
+  unstacked_output_type operator()(const std::vector<Tensor>& step_inputs, const hidden_type& input_hidden, const cell_params& params, bool reverse = false) const {
+    std::vector<Tensor> step_outputs(step_inputs.size());
     auto hidden = input_hidden;
-    for (size_t i = 0; i < step_inputs.size(); i++) {
-      hidden = cell_(step_inputs[i], hidden, params);
-      step_outputs.push_back(hidden_as_output(hidden));
+    for (size_t idx = 0; idx < step_inputs.size(); idx++) {
+      size_t input_idx = reverse ? step_inputs.size() - 1 - idx : idx;
```

Comment:
Does this actually help in benchmarks?

Reply:
Mostly just curious

Reply:
it seems it's about the same speed or slightly faster based on benchmark